connectivity/FEATURE_BLE/include/ble/SecurityManager.h

```diff
@@ -861,171 +909,39 @@ class SecurityManager {
-
-    ble_error_t allowLegacyPairing_(bool allow);
-
-    ble_error_t getSecureConnectionsSupport_(bool *enabled);
-
-    ble_error_t setIoCapability_(SecurityIOCapabilities_t iocaps);
-
-    ble_error_t setDisplayPasskey_(const Passkey_t passkey);
-
-    ble_error_t setLinkSecurity_(
-        ble::connection_handle_t connectionHandle,
-        SecurityMode_t securityMode
-    );
-
-    ble_error_t setKeypressNotification_(bool enabled);
-
-    ble_error_t enableSigning_(
-        ble::connection_handle_t connectionHandle,
-        bool enabled
-    );
-
-    ble_error_t setHintFutureRoleReversal_(bool enable);
-
-    ble_error_t getLinkEncryption_(
-        ble::connection_handle_t connectionHandle,
-        ble::link_encryption_t *encryption
-    );
-
-    ble_error_t setLinkEncryption_(
-        ble::connection_handle_t connectionHandle,
-        ble::link_encryption_t encryption
-    );
-
-    ble_error_t setEncryptionKeyRequirements_(
-        uint8_t minimumByteSize,
-        uint8_t maximumByteSize
-    );
-
-    ble_error_t requestAuthentication_(
-        ble::connection_handle_t connectionHandle
-    );
-
-    ble_error_t generateOOB_(const ble::address_t *address);
-
-    ble_error_t setOOBDataUsage_(
-        ble::connection_handle_t connectionHandle,
-        bool useOOB,
-        bool OOBProvidesMITM
-    );
-
-    ble_error_t confirmationEntered_(
-        ble::connection_handle_t connectionHandle,
-        bool confirmation
-    );
-
-    ble_error_t passkeyEntered_(
-        ble::connection_handle_t connectionHandle,
-        Passkey_t passkey
-    );
-
-    ble_error_t sendKeypressNotification_(
-        ble::connection_handle_t connectionHandle,
-        Keypress_t keypress
-    );
-
-    ble_error_t legacyPairingOobReceived_(
-        const ble::address_t *address,
-        const ble::oob_tk_t *tk
-    );
-
-    ble_error_t oobReceived_(
-        const ble::address_t *address,
-        const ble::oob_lesc_value_t *random,
-        const ble::oob_confirm_t *confirm
-    );
-
-    ble_error_t getSigningKey_(
-        ble::connection_handle_t connectionHandle,
-        bool authenticated
-    );
-
-private:
-    SecurityManagerShutdownCallbackChain_t shutdownCallChain;
-
-protected:
-    EventHandler*      eventHandler;
-    EventHandler       defaultEventHandler;
+public:
+#if !defined(DOXYGEN_ONLY)
+    /** For backwards compatibility. This enm is now in BLETypes.h
+     * @deprecated, use the enum in ble namespace */
+    enum Keypress_t {
+        KEYPRESS_STARTED,   /**< Passkey entry started */
+        KEYPRESS_ENTERED,   /**< Passkey digit entered */
+        KEYPRESS_ERASED,    /**< Passkey digit erased */
+        KEYPRESS_CLEARED,   /**< Passkey cleared */
+        KEYPRESS_COMPLETED, /**< Passkey entry completed */
+    };
+#endif // !defined(DOXYGEN_ONLY)
 };
 
-
 #if !defined(DOXYGEN_ONLY)
-} // interface
+} // namespace interface
+#endif // !defined(DOXYGEN_ONLY)
 } // ble
 
-using ble::impl::SecurityManager;
```

Comment:
I thought Gap and the rest are supposed to be in ble:: namespace, looking at the old files I don't see a using ble::Gap

Examples don't compile because they were incorrectly failing to include headers and relying on the happy coincidence of other headers including what they needed, I have made the fixes but not pushed to master yet

Reply:
I added it as a config option